test/livebook/intellisense/docs_test.exs

```diff
@@ -0,0 +1,47 @@
+defmodule Livebook.Intellisense.DocsTest do
+  use ExUnit.Case, async: true
+
+  alias Livebook.Intellisense.Docs
+
+  @code ~S'''
+  defmodule GoToDefinition do
+    @type t :: term()
+    @type foo :: foo(:bar)
+    @type foo(var) :: {var, t()}
+
+    defmacro with_logging(do: block) do
+      quote do
+        require Logger
+        Logger.debug("Running code")
+        result = unquote(block)
+        Logger.debug("Result: #{inspect(result)}")
+        result
+      end
+    end
+
+    @spec hello(var :: term()) :: foo(term())
+    def hello(message) do
+      {:bar, message}
+    end
+  end
+  '''
+
+  @tag :tmp_dir
+  test "locate_definition/2 returns the definition's location with cell id and line",
```

Comment:
I think it would be better to have this as intellisense test, we define/write the module to the `context.ebin_path` and have the same line assertions but against the generated docs string. We do that for all other intellisense stuff. The reason is that what really matters is the intellisense response, how we extract it is an implementation detail (even if in a separate module). We could obviously test in both places, but I don't think we need in this case :)